mongo/cursor.go

```diff
@@ -67,6 +68,45 @@ func newEmptyCursor() *Cursor {
 	return &Cursor{bc: driver.NewEmptyBatchCursor()}
 }
 
+// NewMockCursor creates a new Cursor pre-loaded with the provided documents, error and registry. If no registry is provided,
+// bson.DefaultRegistry will be used.
+//
+// The documents parameter must be a slice of documents.
```

Comment:
I suggest noting requirements of `documents`, similar to the documentation for `collection#InsertMany`.
```suggestion
// The documents parameter must be a slice of documents. The slice may be nil or empty.
// The elements must all be non-nil.
```

Reply:
Good point; documented.